cptvtool/main.go

```diff
@@ -21,37 +35,39 @@ func runMain() error {
 	if len(os.Args) != 2 {
 		return fmt.Errorf("usage: %s <filename>", os.Args[0])
 	}
-	file, err := os.Open(os.Args[1])
-	if err != nil {
-		return err
-	}
-	defer file.Close()
-	bfile := bufio.NewReader(file)
-	r, err := cptv.NewParser(bfile)
-	if err != nil {
-		return err
-	}
-	fields, err := r.Header()
+
+	fr, err := cptv.NewFileReader(os.Args[1])
 	if err != nil {
 		return err
 	}
+	defer fr.Close()
 
-	fmt.Println(fields.Timestamp(cptv.Timestamp))
-	fmt.Println(fields.Uint32(cptv.XResolution))
-	fmt.Println(fields.Uint32(cptv.YResolution))
-	fmt.Println(fields.Uint8(cptv.Compression))
-	fmt.Println(fields.String(cptv.DeviceName))
+	fmt.Println("Timestamp:   ", fr.Timestamp())
+	fmt.Println("Device Name: ", fr.DeviceName())
 
+	// Read the frames and get a frame count. This is an illustration of
+	// frame reading - the r.FrameCount method will do the same thing (and
+	// will similarly leave the file pointer at EOF)
 	frames := 0
+	var frame lepton3.Frame
```

Comment:
A common pattern when a type is usually handled as a pointer is to use new: 

```go
frame := new(lepton3.Frame)
```

This allocates an instance of the type and returns a pointer to it.

What you've done is also fine though. Just more for your information.